### PR TITLE
feat: wait for compute plan & tasks to finish before using output assets

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Related issue

`#` followed by the number of the issue

## Notes

Useful in FL-994

## Please check if the PR fulfills these requirements

- [ ] If necessary, the [changelog](https://github.com/Substra/substra/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification
- For any breaking changes, companion PRs have been opened on the following repositories:
  - [ ] [substra-tests](https://github.com/Substra/substra)
  - [ ] [substrafl](https://github.com/Substra/substrafl)
  - [ ] [substra-documentation](https://github.com/Substra/substra-documentation)

### CODE REVIEW

**Comment by ThibaultFy** [comment]
Why is it a fix ? and why are you changing this behaviour ? It looks like a feature change to discuss with product 😶

File: substra/sdk/backends/local/backend.py [hunk 1]
```diff
@@ -98,8 +98,8 @@ def list_task_input_assets(self, compute_task_key: str) -> List[models.InputAsse
 
         return inputs
 
-    def get_performances(self, key):
-        performances = self._db.get_performances(key)
+    def get_performances(self, compute_plan: models.ComputePlan):
```
Comment (SdgJlbl) [inline substra/sdk/backends/local/backend.py:101]: Nitpicking, but if we are improving signature typing, could you add the return value too?

File: substra/sdk/backends/local/dal.py [hunk 2]
```diff
@@ -104,14 +104,13 @@ def get(self, type_, key: str):
                 return self._remote.get(type_, key)
             raise
 
-    def get_performances(self, key):
+    def get_performances(self, compute_plan: models.ComputePlan):
```
Comment (SdgJlbl) [inline substra/sdk/backends/local/dal.py:107]: Typing return too :)

File: substra/sdk/backends/remote/backend.py [hunk 3]
```diff
@@ -74,11 +74,12 @@ def list_task_input_assets(self, compute_task_key: str) -> List[models.InputAsse
         )
         return [models.InputAsset(**i) for i in inputs]
 
-    def get_performances(self, key):
+    def get_performances(self, compute_plan: models.ComputePlan):
```
Comment (SdgJlbl) [inline substra/sdk/backends/remote/backend.py:77]: ect 🙃

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-- `wait_task` and `wait_compute_plan` to block execution until execution is over ([#368](https://github.com/Substra/substra/pull/368))
+- `wait_completion` param on `wait_task` and `wait_compute_plan` to block execution until execution is over ([#368](https://github.com/Substra/substra/pull/368))
```
Comment (ThibaultFy) [inline CHANGELOG.md:12]: The param is not on `wait_task` and `wait_compute_plan` is it ?

File: CHANGELOG.md [hunk 5]
```diff
@@ -20,6 +20,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Pin `pydantic` to `>=1.9.0` & `<2.0.0` as `pydantic` v `2.0.0` has been released with a lot of non backward compatible changes. ([#371](https://github.com/Substra/substra/pull/371))
 
+### Fixed
+
+- `list_task_output_assets` and `get_task_output_asset` wait that the compute task is over before getting assets ([#369](https://github.com/Substra/substra/pull/369))
+- `get_performances` wait that compute plan is over before getting assets ([#369](https://github.com/Substra/substra/pull/369))
+
```
Comment (ThibaultFy) [inline CHANGELOG.md:27]: Duplicate of the fixed section. And is it really a fix ? Looks like a feature no ?

**Review by ThibaultFy — COMMENTED** [review]
Thanks for the PR, just a few comment on the changelog

File: CHANGELOG.md (hunk 5 above)
Comment (guilhem-barthes) [reply CHANGELOG.md:27]: @RomainGoussault What is your opinion?

**Review by ThibaultFy — APPROVED** [review]
thanks :)

**Pull request merged**